addon/services/popover.js

```diff
@@ -1,12 +1,99 @@
 import Ember from 'ember';
+import { assert } from '@ember/debug';
+import ModalComponent from 'ember-widgets/components/modal-component';
+import PopoverComponent from 'ember-widgets/components/popover-component';
+import $ from 'jquery';
+
+function classInheritsFrom(childKlass, parentKlass) {
+  return parentKlass.prototype.isPrototypeOf(childKlass.prototype);
+}
 
 export default Ember.Service.extend({
   init() {
     this._super(...arguments);
+
+    // Array of specifications for modals and/or popovers
     this.popovers = Ember.A([]);
   },
 
-  open(destinationElement, ComponentClass, options) {
+  openPopover(componentName, options={}, hideAll=true) {
+    assert(`[ember-widgets#openPopover] {{render-popover}} must be included in a template for popovers to open`, !!this._popoverDidRender);
+
+    let ComponentClass = this.container.lookupFactory(`component:${componentName}`);
+    assert(`[ember-widget#openPopover] cannot find component class for name "${componentName}"`,
+            !!ComponentClass,);
+    assert(`[ember-widgets#PopoverComponent] The Component Class for "${componentName}" does not extend PopoverComponent`,
+            classInheritsFrom(ComponentClass, PopoverComponent));
+    
+    let rootElementSelector = options.rootElement;
+    delete options.rootElement;
+    let destinationElement = this._getDestinationElement(ComponentClass, rootElementSelector);
+
+    let closePopover = () => {
+      this.popovers.removeObject(popoverSpec);
+    };
+    let popoverSpec = {
+      componentName,
+      destinationElement,
+      propertyOptions: {
+        ...options, closePopover
+      }
+    };
+
+    if (hideAll) {
+      this._hideAllPopovers();
+    }
+    this.popovers.pushObject(popoverSpec);
+    return closePopover;
+  },
+
+  openModal(componentName, options={}) {
+    assert(`[ember-widgets#openModal] {{render-popover}} must be included in a template for popovers to open`, !!this._popoverDidRender);
+
+    let ComponentClass = this.container.lookupFactory(`component:${componentName}`);
```

Comment:
Not sure if modals are also occasionally views, but worth checking.

Reply:
Checked the app myself, we're all good 😄

Reply:
thanks for checking!